File: src/main/java/luminia/backend/models/Course.java
package luminia.backend.models;

import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity(name = "Course")
@Table(name = "courses")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Course {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "courses_gen")
    @SequenceGenerator(name = "courses_gen", sequenceName = "courses_seq", allocationSize = 1)
    @Column(name = "id", nullable = false)
    private Long id;
    private String name;
    private String description;
    @ManyToOne
    @JoinColumn(name = "teacher_id")
    private User teacher;
    @OneToMany(mappedBy = "course", fetch = FetchType.LAZY)
    private List<Task> tasks;
}

File: frontend/src/services/remote/course.ts
import type User from './user'

export default class Course {
    public id: number
    public name: string
    public description: string
    public teacher: User
    public status: string

    constructor(
        id: number,
        name: string,
        description: string,
        teacher: User,
        status: string
    ) {
        this.id = id
        this.name = name
        this.description = description
        this.teacher = teacher
        this.status = status
    }
}